core/solidus_core.gemspec

```diff
@@ -29,8 +29,8 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'activemerchant', '~> 1.66'
   s.add_dependency 'acts_as_list', '~> 0.3'
-  s.add_dependency 'awesome_nested_set', '~> 3.0', '>= 3.0.1'
-  s.add_dependency 'cancancan', '~> 2.2'
+  s.add_dependency 'awesome_nested_set', '~> 3.2', '>= 3.0.1'
+  s.add_dependency 'cancancan', '~> 3.0'
```

Comment:
There is no need to use 3 and above. It also works with `['>= 2.1', '< 4.0']`
We want to keep our dependencies always as alow as possible to avoid conflicts with other gems.

Reply:
@tvdeyen I tried with >= 2.1, < 4.0, but I get tons of these:
```
 111) Taxonomies show should display existing taxonomies
       Failure/Error: @search = ~Spree~::~Order~.accessible_by(current_ability, :index).ransack(params[:q])
       
       CanCan::Error:
         The accessible_by call cannot be used with a block ‘can’ definition.The SQL cannot be determined for :index Spree::Order
```

Reply:
Interesting. This does not seem to be Rails 6 related to me. :thinking:

Reply:
I think we need https://github.com/solidusio/solidus/pull/3174 as well? Maybe loading that code conditionally if Rails >= 6

Reply:
With CanCanCan 3, everything seems to work without #3174. Not sure if #3174 would allow us to continue using CanCanCan 2?

Reply:
I commented on #3174 that we should try to allow 2 as well. It should work as it only removes lots of quirks.

Reply:
Even with the patch from #3174, I keep seeing lots of test failures when using CanCanCan 2. I think an upgrade to 3 is required, but again, let me know if you find a way to make the suite pass.

Reply:
I would like to see some of these errors. I ask myself what a Rails update should change in terms of these errors from cancancan. The issue is that we try to use the same ability rule for scopes as well for single records. Then cancancan is not able to generate the correct sql for the scope. Very interesting. 

I really would like to not raise a commonly used dependency like cancancan if we find a way to not need it. I know this might be disappointing and we all want to land Rails 6 support, but we also have responsibility for not breaking extensions and stores that might use dependencies that are not supporting cancancan 3 yet.